### PR TITLE
mariadb 10.7.3

### PR DESCRIPTION
**Pull request by SMillerDev**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Fun:
```
   /tmp/mariadb-20220214-34217-5hhc2l/mariadb-10.7.3/client/mysql.cc:2747:59: error: expected expression
    rl_attempted_completion_function= (rl_completion_func_t*)&new_mysql_completion;
                                                            ^
  /tmp/mariadb-20220214-34217-5hhc2l/mariadb-10.7.3/client/mysql.cc:2747:38: error: use of undeclared identifier 'rl_completion_func_t'; did you mean 'rl_completion_matches'?
    rl_attempted_completion_function= (rl_completion_func_t*)&new_mysql_completion;
                                       ^~~~~~~~~~~~~~~~~~~~
                                       rl_completion_matches
  /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/editline/readline.h:202:16: note: 'rl_completion_matches' declared here
  char           **rl_completion_matches(const char *, rl_compentry_func_t *);
                   ^
  /tmp/mariadb-20220214-34217-5hhc2l/mariadb-10.7.3/client/mysql.cc:2748:33: error: incompatible function pointer types assigning to 'Function *' (aka 'int (*)(const char *, int)') from 'rl_compentry_func_t *' (aka 'char *(*)(const char *, int)')
    rl_completion_entry_function= (rl_compentry_func_t*)&no_completion;
                                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
  /tmp/mariadb-20220214-34217-5hhc2l/mariadb-10.7.3/client/mysql.cc:2750:3: error: no matching function for call to 'rl_add_defun'
    rl_add_defun("magic-space", (rl_command_func_t *)&fake_magic_space, -1);
    ^~~~~~~~~~~~
  /Library/Developer/CommandLineTools/SDKs/MacOSX12.sdk/usr/include/editline/readline.h:195:7: note: candidate function not viable: no known conversion from 'rl_command_func_t *' (aka 'int (*)(int, int)') to 'Function *' (aka 'int (*)(const char *, int)') for 2nd argument
  int              rl_add_defun(const char *, Function *, int);
                   ^
  4 errors generated.
```

**Comment by makigumo** [comment]
That's the same issue as with latest 10.6.x releases: https://github.com/Homebrew/homebrew-core/pull/95021 https://github.com/Homebrew/homebrew-core/pull/94790

**Comment by SMillerDev** [comment]
Someone actually had a talk on fosdem this year indicating that mariadb isn't actually tested on macOS so this is going to be fun.

**Comment by Bo98** [comment]
Removing `-DWITH_READLINE=yes` should be sufficient here (and adding a readline dependency for Linux).

**Comment by Bo98** [comment]
Shouldn't need the patch I reckon, but I haven't tested locally.

**Comment by carlocab** [comment]
```
-- Performing Test HAVE_SYSTEM_LIBFMT
-- Performing Test HAVE_SYSTEM_LIBFMT - Failed
```
And no linkage with `fmt`. Having a look at how to fix it.

**Comment by carlocab** [comment]
Ok, that didn't work. Let's try a patch...

File: Formula/mariadb.rb [hunk 1]
```diff
@@ -47,11 +49,15 @@ class Mariadb < Formula
...
   conflicts_with "mysql", "percona-server",
     because: "mariadb, mysql, and percona install the same binaries"
+
   conflicts_with "mytop", because: "both install `mytop` binaries"
   conflicts_with "mariadb-connector-c", because: "both install `mariadb_config`"
 
   fails_with gcc: "5"
 
+  # Fix finding Homebrew `fmt`
+  patch :DATA
```
Comment (carlocab) [inline Formula/mariadb.rb:59]: Will upstream this if it works.

**Comment by carlocab** [comment]
```
  -- Performing Test HAVE_SYSTEM_LIBFMT
  -- Performing Test HAVE_SYSTEM_LIBFMT - Success
```
Unfortunately, something else broke. 😭

**Comment by SMillerDev** [comment]
It doesn't seem to be one of the [known issues](https://jira.mariadb.org/browse/MDEV-27758?jql=project%20%3D%20MDEV%20AND%20resolution%20%3D%20Unresolved%20AND%20labels%20%3D%20macos%20ORDER%20BY%20priority%20DESC%2C%20updated%20DESC) either.

**Comment by carlocab** [comment]
Well, let's see how it does on Linux...

**Comment by SMillerDev** [comment]
At least it worked there

**Comment by cho-m** [comment]
Linux isn't showing linkage to `fmt` or `readline`
```
Dependencies with no linkage:
  fmt
  readline
```

**Comment by carlocab** [comment]
Could be using `fmt` for the headers only. The check for system `fmt` also tests only the headers. Not sure about `readline`.

**Comment by Bo98** [comment]
> Could be using `fmt` for the headers only. The check for system `fmt` also tests only the headers.

Yeah it's using `FMT_HEADER_ONLY` I think.

**Comment by cho-m** [comment]
If you still want to use Homebrew's `fmt`, you might be able to get past error with `ENV.cxx11` for macOS/Clang.

**Comment by carlocab** [comment]
> If you still want to use Homebrew's `fmt`, you might be able to get past error with `ENV.cxx11` for macOS/Clang.

I remember thinking there was some issue with `ENV.cxx11`. But now I don't remember what it is, so let's try it...

**Comment by SMillerDev** [comment]
Looks good to me.

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**